shared/protocol/src/main/java/io/pravega/shared/protocol/netty/AppendDecoder.java

```diff
@@ -95,29 +95,43 @@ public Request processCommand(WireCommand command) throws Exception {
                     ca.getExpectedOffset(), ca.getRequestId());
             break;
         case APPEND_BLOCK:
-            getSegment(((WireCommands.AppendBlock) command).getWriterId());
             currentBlock = (WireCommands.AppendBlock) command;
+            getSegment(currentBlock.getWriterId());
             result = null;
             break;
         case APPEND_BLOCK_END:
             WireCommands.AppendBlockEnd blockEnd = (WireCommands.AppendBlockEnd) command;
-            if (currentBlock == null) {
-                log.warn("Received AppendBlockEnd {} without AppendBlock", blockEnd);
-                throw new InvalidMessageException("AppendBlockEnd without AppendBlock.");
-            }
             UUID writerId = blockEnd.getWriterId();
-            if (!writerId.equals(currentBlock.getWriterId())) {
-                throw new InvalidMessageException("AppendBlockEnd for wrong connection.");
-            }
             segment = getSegment(writerId);
+            int sizeOfWholeEventsInBlock = blockEnd.getSizeOfWholeEvents();
+            ByteBuf appendDataBuf;
+            if (blockEnd.numEvents <= 0) {
+                throw new InvalidMessageException("Invalid number of events in block. numEvents : " + blockEnd.numEvents);
+            }
             if (blockEnd.getLastEventNumber() < segment.lastEventNumber) {
-                throw new InvalidMessageException("Last event number went backwards.");
+                throw new InvalidMessageException("Last event number went backwards." +
```

Comment:
Nit pick: Let's use `String.format`; it makes the code easier to read.

Reply:
yes. its done